lib/logstash/inputs/log4j.rb

```diff
@@ -59,7 +57,25 @@ def register
     @logger.info("Log4j input")
   end # def register
 
-  public
+  def run(output_queue)
+    if server?
+      while !stop?
+        Thread.start(@server_socket.accept) do |s|
+          s.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
```

Comment:
either way, I just thought it would have some benefit to mention the method cache bit.  This was discussed at length by some of the Ruby community when DCI vs delegation was a hot topic.